base/compiler/abstractinterpretation.jl

```diff
@@ -639,11 +639,25 @@ function pure_eval_call(@nospecialize(f), argtypes::Vector{Any}, @nospecialize(a
     end
 end
 
+function argtype_by_index(argtypes::Vector{Any}, i::Int)
+    n = length(argtypes)
+    if isvarargtype(argtypes[n])
+        return i >= n ? unwrapva(argtypes[n]) : argtypes[i]
+    else
+        return i > n ? Bottom : argtypes[i]
+    end
+end
+
 function abstract_call(@nospecialize(f), fargs::Union{Nothing,Vector{Any}}, argtypes::Vector{Any}, vtypes::VarTable, sv::InferenceState, max_methods = sv.params.MAX_METHODS)
     if f === _apply
-        return abstract_apply(nothing, argtypes[2], argtypes[3:end], vtypes, sv, max_methods)
+        ft = argtype_by_index(argtypes, 2)
+        ft === Bottom && return Bottom
+        return abstract_apply(nothing, ft, argtypes[3:end], vtypes, sv, max_methods)
```

Comment:
If index 2 can be bad, that seems to imply that 3:end would be bad too (vararg case)?

Reply:
It's ok, it just gives an empty array.

Reply:
right—that seems it'd be bad if `argtypes = [f, Vararg{t}]`, since the slice should give `[Vararg{t}]`

Reply:
I think that would probably be fine actually. I think the problematic part is `argtypes = [Vararg{t}]` since then `argtypes[3:end]` is `[]`, but clearly there are args there.